Lib/sysconfig.py

```diff
@@ -544,16 +544,20 @@ def _init_posix(vars):
 def _init_non_posix(vars):
     """Initialize the module as appropriate for NT"""
     # set basic install directories
-    import _imp
+    import _winapi
+    import _sysconfig
     vars['LIBDEST'] = get_path('stdlib')
     vars['BINLIBDEST'] = get_path('platstdlib')
     vars['INCLUDEPY'] = get_path('include')
-    try:
-        # GH-99201: _imp.extension_suffixes may be empty when
-        # HAVE_DYNAMIC_LOADING is not set. In this case, don't set EXT_SUFFIX.
-        vars['EXT_SUFFIX'] = _imp.extension_suffixes()[0]
-    except IndexError:
-        pass
+
+    # Add EXT_SUFFIX, SOABI, and Py_NOGIL
+    vars.update(_sysconfig.config_vars())
```

Comment:
(note for the future)

I think we can move this to `_init_config_vars`, as I am sure it will end up there soon anyway. The only thing to keep in mind is that we allow the POSIX config vars module (eg. `_sysconfigdata__linux_x86_64-linux-gnu`) to be overwritten (to help in cross-compilation) by setting `_PYTHON_SYSCONFIGDATA_NAME`, and we should raise an error if the value given by the user is different from what we computed.
To be even more cautious, I think we should probably save the original `sysconfigdata` name in the C module, as attributes like `sys.implementation`, etc., are often monkey patched when cross-compiling. If needed, people working on cross-compilation should also monkey patch the `_sysconfig` module explicitly, or better, just monkeypatch `sysconfig.get_config_vars`.

You don't need to do it in this PR, I just wanted to write this down, especially to document the non-obvious issue with `_PYTHON_SYSCONFIGDATA_NAME`.
I don't think relying on Makefile variables is great, for a ton of reasons, meaning the new sysconfig API (GH-103480) will very likely use `_sysconfig.config_vars` on other platforms, so I expect the `_PYTHON_SYSCONFIGDATA_NAME` issue to be relevant soon.